pkg/model/iam.go

```diff
@@ -164,11 +165,13 @@ func (b *IAMModelBuilder) buildIAMRolePolicy(role iam.Subject, iamName string, i
 		},
 	}
 
-	// This is slightly tricky; we need to know the hosted zone id,
-	// but we might be creating the hosted zone dynamically.
-	// We create a stub-reference which will be combined by the execution engine.
-	iamPolicy.DNSZone = &awstasks.DNSZone{
-		Name: fi.String(b.NameForDNSZone()),
+	if !dns.IsGossipHostname(b.Cluster.ObjectMeta.Name) {
+		// This is slightly tricky; we need to know the hosted zone id,
+		// but we might be creating the hosted zone dynamically.
+		// We create a stub-reference which will be combined by the execution engine.
+		iamPolicy.DNSZone = &awstasks.DNSZone{
+			Name: fi.String(b.NameForDNSZone()),
+		}
```

Comment:
This brings up an interesting question: should we be disabling the dns-controller addon when using gossip? I looks like we might be running it with IAM over no DNS zones?

Reply:
I created https://github.com/kubernetes/kops/issues/9915 so that we can discuss about it.